tensorrt_llm/_torch/compilation/backend.py

```diff
@@ -37,7 +37,7 @@ def __init__(
         enable_inductor=True,
         enable_userbuffers=False,
         enable_piecewise_cuda_graph: bool = False,
-        cuda_graph_batch_sizes: Optional[List[int]] = None,
+        capture_num_tokens: List[int] = [],
```

Comment:
The list being the default value is problematic
> because default values are evaluated only once at function definition time, not each time the function is called.
> That means the same list instance is shared across all calls to the function where that argument is not explicitly passed.

The classical solution is to leave the default value as `None`, and then set a `[]` in the function body.